server/core_friend.go

```diff
@@ -18,29 +18,73 @@ import (
 	"database/sql"
 	"errors"
 
+	"encoding/json"
+	"fmt"
+	"github.com/satori/go.uuid"
 	"go.uber.org/zap"
 )
 
-func friendAdd(logger *zap.Logger, db *sql.DB, userID []byte, friendID []byte) error {
+func friendAdd(logger *zap.Logger, db *sql.DB, ns *NotificationService, userID []byte, handle string, friendID []byte) error {
 	tx, txErr := db.Begin()
 	if txErr != nil {
 		return txErr
 	}
 
+	isFriendAccept := false
+	updatedAt := nowMs()
 	var err error
 	defer func() {
 		if err != nil {
 			if rollbackErr := tx.Rollback(); rollbackErr != nil { // don't override value of err
 				logger.Error("Could not rollback transaction", zap.Error(rollbackErr))
 			}
 		} else {
-			if err = tx.Commit(); err != nil {
-				logger.Error("Could not commit transaction", zap.Error(err))
+			if e := tx.Commit(); e != nil {
+				logger.Error("Could not commit transaction", zap.Error(e))
+				err = e
+				return
+			}
+
+			// If the operation was successful, send a notification.
+			content, e := json.Marshal(map[string]interface{}{"handle": handle})
+			if e != nil {
+				logger.Warn("Failed to send friend add notification", zap.Error(e))
+				return
+			}
+			var recipient []byte
+			var sender []byte
+			var subject string
+			var code int64
+			if isFriendAccept {
+				recipient = userID
+				sender = friendID
+				subject = fmt.Sprintf("%v accepted your friend request", handle)
+				code = 3
```

Comment:
Can we use some sort of an Enum here to indicate the message code, so it's not a magic number?